fusesoc/signature.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+# Copyright FuseSoC contributors
+# Licensed under the 2-Clause BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-2-Clause
+
+import base64
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import yaml
+
+from fusesoc import utils
+
+logger = logging.getLogger(__name__)
+
+
+def read_core(data_file_name):
```

Comment:
Let's put this function into the core object instead (and call it someting like get_data_to_sign, but preferably catchier :) ). That way the core itself is responsible for presenting the data to be signed, which gives us a bit more freedom to change things in future CAPI versions.